string.c

```diff
@@ -3602,24 +3602,26 @@ str_uplus(VALUE str)
 
 /*
  * call-seq:
- *   -string -> frozen_string
- *   dedup -> frozen_string
+ *   -self -> frozen_string
  *
- * Returns a frozen, possibly pre-existing copy of the string.
```

Comment:
I think it would be good to keep the "possibly pre-existing" part.

I like your tweaks in general, but I think it makes it much less clear that it helps deduplicate strings.

Reply:
Note added.